include/nbl/asset/filters/CBlitImageFilter.h

```diff
@@ -529,19 +638,28 @@ class CBlitImageFilter : public CImageFilter<CBlitImageFilter<Swizzle,Dither,Nor
 								for (auto h=0; h<MaxChannels; h++)
 									value[h] += windowSample[h];
 							};
-							// do the filtering 
+							// do the filtering
 							core::vectorSIMDf tmp;
 							tmp[axis] = float(i)+0.5f;
 							core::vectorSIMDi32 windowCoord(0);
 							windowCoord[axis] = kernel.getWindowMinCoord(tmp*fScale,tmp)[axis];
 							auto relativePos = tmp[axis]-float(windowCoord[axis]);
+							const uint32_t phaseIndex = i % phaseCount[axis];
```

Comment:
modulo is expensive (several hundred cycles), can you hoist it to above the `for` on line 623
```c
uint32_t phaseIndex = 0;
```
and then at the end of the loop do
```c
if (++phaseIndex==phaseCount[axis])
   phaseIndex = 0;
```

Reply:
Resolved in #343. Except the `assert` because of reasons mentioned [here](https://github.com/Devsh-Graphics-Programming/Nabla/pull/140#discussion_r865541796).

Reply:
ah see, if you would have initialized the `phaseCount` to 1 for any axes that won't be filtered then no assert needed.

You start with a phaseIndex = 0, increment to 1, end up equaling 1, go back to 0

Reply:
maybe don't change the def of `getPhaseCount` but `min(,1)` the `phaseCount you compute here.

Reply:
Resolved.